streams/client/client.go

```diff
@@ -61,101 +111,75 @@ func (c *Client) Context() context.Context {
 	return c.ctx
 }
 
-func New(ctx context.Context, node peer.Node, peers []string, path string, min int, max int) (*Client, error) {
+func New(node peer.Node, path string) (*Client, error) {
 	c := &Client{
 		node: node,
 		path: path,
 	}
 
-	c.ctx, c.ctxC = context.WithCancel(ctx)
+	c.ctx, c.ctxC = context.WithCancel(node.Context())
 
 	return c, nil
 }
 
-func (c *Client) SendTo(cid cid.Cid, cmdName string, body command.Body) (cr.Response, error) {
-	cmd := command.New(cmdName, body)
-	pid, err := peerCore.FromCid(cid)
-	if err != nil {
-		return nil, fmt.Errorf("decoding peer id failed with: %w", err)
-	}
-
-	ctx, ctx_close := context.WithTimeout(c.ctx, SendToPeerTimeout)
-	now := time.Now()
-	defer ctx_close()
-
-	strm, err := c.node.Peer().NewStream(ctx, pid, protocol.ID(c.path))
-	if err != nil {
-		return nil, fmt.Errorf("peer new stream failed with: %w", err)
-	}
-	defer strm.Reset()
-
-	if err = strm.SetWriteDeadline(now.Add(SendTimeout)); err != nil {
-		return nil, fmt.Errorf("set write deadline failed with: %w", err)
+func (c *Client) New(cmd string, opts ...Option) *Request {
```

Comment:
My opinion c.Request(...) sounds better... but either way is fine

Reply:
New like net/http